src/frontends/pytorch/src/op/unfold.cpp

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/frontend/pytorch/node_context.hpp"
+#include "openvino/opsets/opset10.hpp"
+
+using namespace ov::opset10;
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+OutputVector translate_unfold(NodeContext& context) {
+    // constants
+    auto const_0 = context.mark_node(Constant::create(element::i64, Shape{}, {0}));
+    auto const_1 = context.mark_node(Constant::create(element::i64, Shape{}, {1}));
+    auto const_0_list = context.mark_node(Constant::create(element::i64, Shape{1}, {0}));
+    auto const_1_list = context.mark_node(Constant::create(element::i64, Shape{1}, {1}));
+    auto const_neg_1_list = context.mark_node(Constant::create(element::i64, Shape{1}, {-1}));
+
+    // get inputs and prepare auxiliary nodes
+    auto input = context.get_input(0);
+    auto input_shape = context.mark_node(std::make_shared<ShapeOf>(input));
+    auto input_rank = context.mark_node(std::make_shared<ShapeOf>(input_shape));
+
+    auto dimension = context.mark_node(Constant::create(element::i64, Shape{1}, {context.const_input<int64_t>(1)}));
+    auto dimension_plus_1 = context.mark_node(std::make_shared<Add>(dimension, const_1_list));
+
+    auto size_scalar = context.mark_node(std::make_shared<Convert>(context.get_input(2), element::i64));
```

Comment:
It looks like there is no reason to cast to `i64`, correct me if I am wrong. All `i64` values here are created by you, `ShapeOf` can return `i32` values. Please try to avoid cast to `i64`.

Reply:
When get those inputs as "non constant inputs" I have to convert them to integer type, because otherwise it doesn't work. I think that this is approach is not a fault, because these parameters can only be integer type.

Reply:
I introduced changes that you proposed.